### PR TITLE
adding conversion to standard mac addr format

### PR DESCRIPTION
**Pull request by ebeahan**
Adding conversion and mocked data updates to support addressing https://github.com/napalm-automation/napalm-ios/issues/106

### CODE REVIEW

File: napalm_ios/ios.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 from napalm_base.exceptions import ReplaceConfigException, MergeConfigException
 from napalm_base.utils import py23_compat
 import napalm_base.constants as C
+import napalm_base.helpers
```
Comment (mirceaulinic) [inline napalm_ios/ios.py:30]: That means the helpers have never been used? Hmm..

**Review by mirceaulinic — APPROVED** [review]
Thanks @ebeahan!

While approving this, your PR reveals that _none_ of the getters return the standard formats.

The rest of them are:

- `get_interfaces`
- `get_lldp_neighbors` and `get_lldp_neighbors_detail`

In addition, the following should use the `ip` helper to format IPv6 addresses:

- `get_bgp_neighbors`
- `get_interfaces_ip`
- `get_ntp_peers`
- `traceroute`
- `ping`

I suggest merging this PR and open issues to not forget about this inconsistency.

**Comment by mirceaulinic** [comment]
I have raised https://github.com/napalm-automation/napalm-ios/issues/111 to keep tracking of the required changes.

**Comment by ktbyers** [comment]
@mirceaulinic We probably need some better way to integrate format validation into the unit tests. 

In other words, if the format is wrong, unit tests should fail.

File: napalm_ios/ios.py (hunk 1 above)
Comment (ktbyers) [inline napalm_ios/ios.py:30]: Why are we calling 'convert' and not just calling helpers.mac()?

I am not seeing any benefit to that (i.e. just adds an unnecessary level of indirection)?

Comment (ebeahan) [reply napalm_ios/ios.py:30]: I took some direction on the helper implementation from the nxos driver. However there's still some inconsistency even within the nxos driver on whether `convert` is called vs calling `helpers.mac()` directly. 

https://github.com/napalm-automation/napalm-nxos/blob/develop/napalm_nxos/nxos.py#L393-L394

I chose to go the `convert` route as it appears to have some additional error-handling built in. 

https://github.com/napalm-automation/napalm-base/blob/develop/napalm_base/helpers.py#L180-L181

Comment (dbarrosop) [reply napalm_ios/ios.py:30]: I don't think the use of the helpers is very consistent... Not very worried about it though. We can fix things as we go along and OC (which should be the way forward) has type checking builtin. I know it sounds very optimistic but we are doing a huge work here so there is no point of despairing : )

Comment (dbarrosop) [reply napalm_ios/ios.py:30]: I think `mac` is a better name too. Readable code is nice : )  If we need the error handling we can decorate the method or call a common function to do it from within `mac` and friends.

Comment (ktbyers) [reply napalm_ios/ios.py:30]: Okay, so @ebeahan can you convert this over to direct use of the helpers.mac() and then it should be ready to go.

Comment (ebeahan) [reply napalm_ios/ios.py:30]: @dbarrosop @ktbyers Thanks guys will do.

**Pull request merged**